site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/controller/AutoGroupsController.java

```diff
@@ -337,9 +337,9 @@ public String showStep4(Model model, @ModelAttribute AutoGroupsForm autoGroupsFo
                 String groupPrefix = autoGroupsForm.getGroupTitleByGroup();
                 int groupNumber = autoGroupsForm.getGroupNumberByGroup();
                 int groupSize = filteredMembers.size() / groupNumber;
-
+                String indexFormat = String.format("%%0%dd", String.valueOf(groupNumber).length());
```

Comment:
Can you explain how this works?

Reply:
Oh, I see: it will result in "%0xd" where x is the number of digits of the number of groups being created. And then you pass this into the formatter below to pad the group titles with the appropriate number of zeroes. Nice! 

Maybe we need a comment to explain what we are trying to achieve here?

Reply:
Makes sense, added a comment.

Reply:
Thanks! Looks good.